server/src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -548,7 +565,10 @@ public ScriptService getScriptService() {
 
     @Override
     public boolean updateMapping(final IndexMetaData currentIndexMetaData, final IndexMetaData newIndexMetaData) throws IOException {
-        return mapperService().updateMapping(currentIndexMetaData, newIndexMetaData);
+        if (mapperService == null) {
+            return false;
```

Comment:
Can we assert the state of `currentIndexMetaData` and `newIndexMetaData`?

Reply:
What would you assert about those?

Reply:
I thought about asserting the index is closed but MapperService is set once so just ignore this.